src/responses/Response.cpp

```
@@ -113,46 +115,64 @@ void Response::_handleGetRequest(const Request& req) {
         if (S_ISDIR(fileStat.st_mode)) {  // Check if it's a directory
             std::string indexPath = path + "/index.html";
             if (stat(indexPath.c_str(), &fileStat) == 0 && !S_ISDIR(fileStat.st_mode)) {
-                std::string content = _readFile(indexPath);
-                if (content.empty()) {
+                // std::string content = _readFile(indexPath);
+				try {
+					setStatus(200);
+					addHeader("Content-Type", _getMimeType(indexPath));
+					generateResponse(indexPath);
+				} catch (Response::FileSystemErrorException &e) {
                     _setError(404);
-                } else {
-                    setStatus(200);
-                    setBody(content);
-                    addHeader("Content-Type", _getMimeType(indexPath));
-                }
+				} catch (Response::ContentLengthException &e) {
+					throw e; //temp
```

Why not return an error page here?